src/derivative_operators/derivative_operator.jl

```diff
@@ -26,6 +26,29 @@ struct DerivativeOperator{T<:Real,N,Wind,T2,S1,S2<:SArray,T3,F} <: AbstractDeriv
     coeff_func              :: F
 end
 
+struct NonLinearDiffusion! end
```

Comment:
if it's a function then it shouldn't be capitalized like a struct, or need the struct at all.